daemons/attrd/pacemaker-attrd.c

```diff
@@ -251,6 +251,7 @@ main(int argc, char **argv)
 
     initialized = true;
 
+    pcmk__cli_init_logging("pacemaker-attrd", args->verbosity);
```

Comment:
Daemons shouldn't call the cli logging function except potentially for non-daemonizing options like help/version

Reply:
Currently schedulerd, pacemakerd, based, execd, and controld all have lines analogous to this one. I hear what you're saying but I don't understand what's different about attrd (and possibly the fencer, as of this PR).

Reply:
I believe only pacemakerd and pacemaker-schedulerd call the function. pacemakerd correctly uses it only conditionally for a non-daemonizing option, but pacemaker-schedulerd is using it incorrectly at the moment

Reply:
```
[reid@laptop pacemaker]$ grep -R pcmk__cli_init_logging daemons/ 2>/dev/null
daemons/schedulerd/pacemaker-schedulerd.c:    pcmk__cli_init_logging("pacemaker-schedulerd", args->verbosity);
daemons/fenced/cts-fence-helper.c:     * pcmk__cli_init_logging is set up to only handle the latter.
daemons/pacemakerd/pacemakerd.c:        pcmk__cli_init_logging("pacemakerd", args->verbosity);
daemons/based/pacemaker-based.c:    pcmk__cli_init_logging("pacemaker-based", args->verbosity);
daemons/execd/cts-exec-helper.c:     * pcmk__cli_init_logging is set up to only handle the latter.
daemons/execd/pacemaker-execd.c:    pcmk__cli_init_logging(EXECD_NAME, args->verbosity);
daemons/controld/pacemaker-controld.c:    pcmk__cli_init_logging("pacemaker-controld", args->verbosity);
```

Reply:
Ah bummer, I was looking at the 2.1 branch. I should have caught those earlier ...

pcmk__cli_init_logging() calls crm_log_init() with arguments that daemons shouldn't use. Hopefully the later crm_log_init() call overrides those but I haven't checked

Reply:
It was from a very recent PR -- see this thread: https://github.com/ClusterLabs/pacemaker/pull/2950#discussion_r1029861851

I was originally using a for-loop and bumping the log level and then `pcmk__cli_init_logging()` was suggested. I was confused about the difference after you replied, and I never actually changed anything.

Reply:
Yeah the logging is a little clunky at the moment. At this point, I'd deprecate crm_bump_log_level() and replace it with a function that takes the log level directly to the desired level. Then we have crm_log_init() plus a single function call instead of a for loop. Calling crm_log_init() after pcmk__cli_init_logging() is less than ideal; the latter will turn on logging to stderr, and the log level will end up being the max of the two calls.

Reply:
> replace it with a function that takes the log level directly to the desired level.

`set_crm_log_level()` seems to do that. I guess the only question is when we want to enable `stderr` (which `crm_bump_log_level()` does on the first call instead of... bumping the log level).

Reply:
I'm gonna pull this out for a separate PR after this and cibadmin are merged.

Reply:
https://projects.clusterlabs.org/T608